### PR TITLE
Allow antennas to be controlled without connection

### PR DESCRIPTION
**Pull request by Pharylon**
By setting ControlAntennaWithoutConnection to True on the configs, it... well... lets you control antennas without a connection. :)  The rest of the probe remains inoperable.

Reasons for allowing this are twofold:

1. When adding RemoteTech to an existing game, it allows you to use existing probes, where the antennas are certainly inactive, and get them connected to the new network.

2. It's a bit more realistic, as in real life probes are designed to attempt to reconnect if they reboot or otherwise lose connection. Of course, this doesn't exactly duplicate that logic, but it's closer than probes going inoperative all the time.

### CODE REVIEW

**Comment by Peppie23** [comment]
You can also fly with a manned vessel to your satellite and activate the antenna by EVA :stuck_out_tongue_winking_eye:

**Comment by Pharylon** [comment]
LOL. Very true!

**Comment by Pharylon** [comment]
Oh, and obviously this is related to [issue #107](https://github.com/RemoteTechnologiesGroup/RemoteTech/issues/107). I'm not saying it's a 100% solution, obviously, since as I can see from the discussion, there's a need for a solution to "lost craft" that doesn't let the player control it, but for players that think it's fun to pretend that their "playing the probe" attempting to phone home, it solves the issue neatly. 

Lastly, it's just a lot more noob friendly. I think a lot of us probably remember how hard this mod is when we start off... yeah. Sometimes you need a safety net! :D

**Comment by erendrake** [comment]
I would say that being more noob friendly is kinda counter to RemoteTech's focus. RemoteTech is a mod famous for being hard and you have to earn an awesome deep space network.

I am convinced that the only solution that makes sense is to have some kind of plan ahead system that lets you setup something like a backup target or some kind of configurable recovery mode. One that you must still setup ahead of time . 

to answer your points from above.

1. You are right that we should have some kind of "have you never had a connection because we just intalled remotetech? ok lets boot up some antenna and see if we can contact kerbin"

2. You will never convince me that this kind of change is more realistic

**Comment by Pharylon** [comment]
Sure, the hardness is part of the fun. But different people find different levels of hard fun. Different strokes for different folks and all. :D  I find the hardness of building a network through planning out my positions, seeding another planetary SOI with some communications probes before I send a lander, etc, all pretty fun. I really like most of RemoteTech!

But I don't think it's particularly fun to lose a craft part way into a launch because I forgot to put an antenna in an action group. Or to lose contact with a probe around Duna forever because it was pointing at ComSat1 (which was around Kerbin), which just exploded for some reason (when, in real life, this would be a recoverable problem). For people (like me) that want a slightly-less-hard difficulty, is RemoteTech just not for them? Are these players you just don't feel RemoteTech should support/cater to?

**Comment by erendrake** [comment]
tl;dr No, you are presenting a false choice by saying that this we either add a config setting or we are a bunch of elitist jerks who want to make something that isnt fun. I belive there is a better way.

------

No i am not saying that RemoteTech has no room for variable difficulty, 
No i am not saying that people who aren't  "teh hardcore" should uninstall remotetech

What i am saying is that RemoteTech has the gift of a simple rule when it comes to communicating with remote craft:

    We MUST ALWAYS have a connection first. 

I am quite serious when i call it a gift, this kind of context is something that few mods have the luxury of.

For me this means that when we are creating the mod we have to be more creative with the tools we give our users. Simply hand waving this problem with a config setting is simply inadequate.

I am a fan of adding a target priority list and/or an emergency routine that the craft will follow if it is without a connection. Because you are right, real world craft have smarts built in to help deal with communication failures.

this is just to give you an idea what i am talking about for a  "priority list". I could see each antenna having, at least at the start, two targets. One would be primary and the other would only be attempted if the first is not available. This secondary, in my mind, would be defaulted to Kerbin and maybe in the first version wouldn't even be changeable. We could use the number of backup targets as a way to scale difficulty better than what we do today or a benefit of advanced tech. 

For an "emergency mode" or the situation where we have a new RT install.  I don't think that it is unreasonable for the craft to say "Oh crap, all of my antenna are off!" and go through its list of antenna and find the one with the longest range, turn it on and have it evaluate its target list. If the above priority list was implemented too i believe it would be ideal because on a new install each craft would wake up. turn on their big dish and try to contact Kerbin. 

I guess what i am saying is that this PR is a victim of having a goal that, on the surface, seems to be laudable, making the mod more fun. When in fact i believe it would make the mod less fun because you know that the cheat is there. So every time I make a mistake I either have to:

1. Quit the game
2. Change a setting
3. Relaunch the game 
4. Fix the craft 
5. Close the game again 
6. Change the setting back
7. Launch the game AGAIN
8. run my mission. 

Or:

1. leave the setting on all the time. 

which do you think many people would choose? Did that make the mod more fun?

Now that i have written all of that, i realize that i am a bit of a grumpy gus today. Sorry for the tirade. I clearly need a juicebox and a nap.

**Comment by Pharylon** [comment]
I totally agree with what you're saying about how nice it'd be to have fallback targets and all that. It sounds like a pretty good solution. On the other hand, this solution is super simple and here it is. Y'know? A bird in the hand is worth more than three in the bush.

But more of an argument (from my perspective) is this: I don't view it as a cheat. It's the way I *want* it to be. I don't think the game should ever make decisions for me. If my antennas turn on and target spontaneously because the probe lost connection... I mean, I guess that's better than a totally lost ship, but is that fun or intuitive from a end-user standpoint? What if it was a planned outage on the far side of a planet and "seek mode" drains electrical charge I needed? Or maybe I as a player just don't like the idea of it performing any actions I didn't initiate. This could be circumvented by creating a "panic" button on the probe so the player has to manually initiate it's emergency-seek.  But, is that really different than the user clicking on the antenna? The end result is the same. The user clicks on the probe GUI to re-initiate connection. 

Plus, in real life, [probes tend to have low-gain omnis that can reach Earth easily](http://mars.jpl.nasa.gov/mro/mission/spacecraft/parts/antennas/). They're not great for transmitting scientific data, but you can use them to point the main dish. That could be simulated too, creating a special antenna that you still need line-of-sight for, but doesn't transmit science... but my gut feeling is that is over-engineering the solution. In the end, the question is really "what's more fun" not necessarily "what's more realistic" (although the two are often related!). 

So I guess in summary I just think this is a pretty simple solution that covers most (but not all) issues with lost probes in a dead-simple way. Plus, it's an *optional* rule, like ignoring the speed of light. So hardcore players can just ignore it.

As for the difficulty in turning it on/off... for people that do view it as a cheat, isn't it kind of good to make it hard to turn on/off? I know I hate the Debug Menu. I've tried to turn off quickloads a few times. But it's just so easy to turn it back on, and I am weak. :D I actually looked for a mod the other day that would disable the debug menu! Though conversely, having a in-game menu to do things like this or change the antenna distance calculations or the speed of light delay might be nice too.

**Comment by Pharylon** [comment]
P.S. You weren't being grumpy. I totally did what you said and presented a false dichotomy and came across more standoffish than I meant. Thanks for the civil response. :)

**Comment by jdmj** [comment]
> This secondary, in my mind, would be defaulted to Kerbin and maybe in the first version wouldn't even be changeable.
> That could be simulated too, creating a special antenna that you still need line-of-sight for, but doesn't transmit science...
Question related to points above: does anyone remember when and why KSC MC omni range has been nerfed?
